### PR TITLE
refactor!: Update the C++ SDK to use the v2 API.

### PR DESCRIPTION
**Pull request by mplooster**
This also includes an update to the specific version of gRPC that CMake will fetch when building the SDK (now set to get v1.33.2).

Also went for a simpler approach to the C++ SDK with much less wrapping around the gRPC API, so that it is very similar to the Go and Python SDKs.

The diff displayed by Github might be confusing, so it might be easier to simply review the new C++ files.

### CODE REVIEW

**Review by jhollowayj — APPROVED** [review]
Seems fine to me

File: grpc/cpp-diatheke/diatheke_client.cpp [hunk 1]
```diff
@@ -67,213 +67,125 @@ std::string Client::diathekeVersion()
...
-    cobaltspeech::diatheke::SessionID request;
-    request.set_session_id(sessionID);
-
-    // We don't set a deadline on this context because we expect
-    // this to be a long-lived stream.
-    std::shared_ptr<grpc::ClientContext> ctx(new grpc::ClientContext);
+    // Set up the server request
+    cobaltspeech::diatheke::SessionInput request;
+    *(request.mutable_token()) = token;
+    request.mutable_text()->set_text(text);
```
[the remark this answers is not in the document]
Comment (mplooster) [reply grpc/cpp-diatheke/diatheke_client.cpp:137]: Yes. It probably looks silly due to naming. In the protobuf definition, the name of this field for the `SessionInput` message is [`text`](https://github.com/cobaltspeech/sdk-diatheke/blob/master/grpc/diatheke.proto#L103). So `mutable_text` is giving me a mutable version of this field, which itself has a field called [`text`](https://github.com/cobaltspeech/sdk-diatheke/blob/master/grpc/diatheke.proto#L138), which is what the `set_text()` portion refers to. What can I say? Naming is hard. 😜

File: grpc/cpp-diatheke/diatheke_tts_stream.h [hunk 2]
```diff
@@ -27,36 +27,62 @@ namespace Diatheke
...
+    TTSStream(const std::shared_ptr<grpc::ClientContext> &ctx,
+              const std::shared_ptr<GRPCReader> &stream);
 
     ~TTSStream();
 
     /*
-     * Wait for the next ASR result from the server to become available.
-     * Returns false when there are no more results to receive. It is thread-
-     * safe to call this method while also calling pushAudio().
+     * Wait for the next chunk of audio data from the server,
```
Comment (renaissanceGeek) [inline grpc/cpp-diatheke/diatheke_tts_stream.h:43]:
```suggestion
     * Waits for the next chunk of audio data from the server,
```

**Review by renaissanceGeek — APPROVED** [review]
Looks great!  I have a suggestion, but if it's too much hassle, I'm OK with the signature the way it is.

(BTW, I know it's just the conventional commit keyword, but seeing `refactor!` as the checkin message does seem appropriate :grin: )

File: grpc/cpp-diatheke/diatheke_asr_stream.cpp [hunk 3]
```diff
@@ -20,46 +20,82 @@
...
+    const std::shared_ptr<cobaltspeech::diatheke::ASRResult> &result,
+    const std::shared_ptr<GRPCWriter> &stream)
+    : mContext(ctx), mResult(result), mStream(stream)
 {
 }
 
 ASRStream::~ASRStream() {}
 
-void ASRStream::pushAudio(const char *audio, size_t sizeInBytes)
+bool ASRStream::sendAudio(const std::string &data)
```
[the remark this answers is not in the document]
Comment (mplooster) [reply grpc/cpp-diatheke/diatheke_asr_stream.cpp:33]: Yes. That is how protobuf represents a field of `bytes` in C++. I suppose that technically a `std::string` is an array of `char*`, so this certainly feels more like C++.

File: grpc/cpp-diatheke/diatheke_asr_stream.h [hunk 4]
```diff
@@ -20,60 +20,94 @@
...
     /*
-     * Push the given audio data to the Diatheke server. The audio format
-     * should match what is specified in the Diatheke server configuration.
-     * If there is no audio data (i.e., sizeInBytes equals zero), the server
-     * will treat this as EOF and end the stream. Subsequent calls to
-     * pushAudio will fail.
+     * Send the given audio data to Diatheke for transcription.
+     *
+     * If this function returns False, the server has closed the stream
+     * and result() should be called to get the final ASR result.
```
[the remark this answers is not in the document]
Comment (mplooster) [reply grpc/cpp-diatheke/diatheke_asr_stream.h:47]: I think my mental model is that the bool represents the state of the stream. `true` = "OK" and `false` = "closed".

**Pull request merged**